src/main/java/com/yelp/nrtsearch/server/luceneserver/StatsRequestHandler.java

```diff
@@ -0,0 +1,139 @@
+        for (Map.Entry<Integer, ShardState> entry : indexState.shards.entrySet()) {
+            ShardState shardState = entry.getValue();
+            statsResponseBuilder.setOrd(entry.getKey());
+            IndexWriter.DocStats docStats = shardState.writer.getDocStats();
+            String[] fNames = shardState.indexDir.listAll();
+            long dirSize = 0;
+            for (int i = 0; i < fNames.length; i++) {
+                dirSize += shardState.indexDir.fileLength(fNames[i]);
+            }
+            statsResponseBuilder.setMaxDoc(docStats.maxDoc);
+            statsResponseBuilder.setNumDocs(docStats.numDocs);
+            statsResponseBuilder.setDirSize(dirSize);
+            // TODO: snapshots
+
+            // TODO: go per segment and print more details, and
+            // only print segment for a given searcher if it's
+            // "new"
+
+            // Doesn't actually prune; just gathers stats
+            List<Searcher> tmpSearchers = new ArrayList<>();
+            shardState.slm.prune(new SearcherLifetimeManager.Pruner() {
+                @Override
+                public boolean doPrune(double ageSec, IndexSearcher indexSearcher) {
+                    Searcher.Builder searcher = Searcher.newBuilder();
+                    searcher.setVersion(((DirectoryReader) indexSearcher.getIndexReader()).getVersion());
+                    searcher.setStaleAgeSeconds(ageSec);
+                    searcher.setSegments(indexSearcher.getIndexReader().toString());
+                    searcher.setNumDocs(indexSearcher.getIndexReader().maxDoc());
+                    tmpSearchers.add(searcher.build());
+                    return false;
+                }
+            });
+            statsResponseBuilder.addAllSearchers(tmpSearchers);
+            statsResponseBuilder.setState(shardState.getState());
+
+            SearcherTaxonomyManager.SearcherAndTaxonomy s = shardState.acquire();
+            try {
+                Taxonomy.Builder taxonomy = Taxonomy.newBuilder();
+                if (s.taxonomyReader != null) { //taxo reader is null for primary and replica
+                    taxonomy.setNumOrds(s.taxonomyReader.getSize());
+                    taxonomy.setSegments(s.taxonomyReader.toString());
+                }
+                statsResponseBuilder.setTaxonomy(taxonomy.build());
+                Searcher.Builder searcher = Searcher.newBuilder();
+                if (s.searcher != null) {
+                    searcher.setSegments(s.searcher.toString());
+                    searcher.setNumDocs(s.searcher.getIndexReader().numDocs());
+                }
+                statsResponseBuilder.setCurrentSearcher(searcher.build());
+            } finally {
+                shardState.release(s);
+            }
+        }
+        return statsResponseBuilder.build();
+
+    }
+
+    public static IndicesResponse getStatsResponse(GlobalState globalState) throws IOException, HandlerException {
```

Comment:
Should this be `getIndicesResponse` ? Should this also be in a separate handler ?

Reply:
the handler API doesnt work well for this once since we get an empty request anyway.